webknossos-datastore/app/com/scalableminds/webknossos/datastore/helpers/DataSetDeleter.scala

```diff
@@ -0,0 +1,38 @@
+package com.scalableminds.webknossos.datastore.helpers
+import com.scalableminds.util.tools.Fox
+import com.typesafe.scalalogging.LazyLogging
+import net.liftweb.common.Full
+
+import java.io.File
+import java.nio.file.{Files, Path}
+import scala.concurrent.ExecutionContext
+
+trait DataSetDeleter extends LazyLogging {
+  def dataBaseDir: Path
+
+  def deleteOnDisk(organizationName: String, dataSetName: String, isInConversion: Boolean = false)(
+      implicit ec: ExecutionContext): Fox[Unit] = {
+    val dataSourcePath =
+      if (isInConversion) dataBaseDir.resolve(organizationName).resolve(".forConversion").resolve(dataSetName)
+      else dataBaseDir.resolve(organizationName).resolve(dataSetName)
+    val trashPath: Path = dataBaseDir.resolve(organizationName).resolve(".trash")
+    val targetPath = trashPath.resolve(dataSetName)
+    new File(trashPath.toString).mkdirs()
+
+    logger.info(s"Deleting dataset by moving it from $dataSourcePath to $targetPath...")
```

Comment:
Maybe add the possibility to pass a “reason” String to this method which is included in the logging. This way, it is easier to reconstruct what happened, i.e. discerning if something was deleted explicitly by a user or by the failure cleanup.